chatsky/context_storages/database.py

```diff
@@ -8,181 +8,156 @@
 This class implements the basic functionality and can be extended to add additional features as needed.
 """
 
-import asyncio
-import importlib
-import threading
-from functools import wraps
 from abc import ABC, abstractmethod
-from typing import Callable, Hashable, Optional
+from asyncio import Lock
+from functools import wraps
+from importlib import import_module
+from pathlib import Path
+from typing import Any, Awaitable, Callable, Dict, List, Literal, Optional, Set, Tuple, Union
 
 from .protocol import PROTOCOLS
-from chatsky.core import Context
-
 
-class DBContextStorage(ABC):
-    r"""
-    An abstract interface for `chatsky` DB context storages.
-    It includes the most essential methods of the python `dict` class.
-    Can not be instantiated.
-
-    :param path: Parameter `path` should be set with the URI of the database.
-        It includes a prefix and the required connection credentials.
-        Example: postgresql+asyncpg://user:password@host:port/database
-        In the case of classes that save data to hard drive instead of external databases
-        you need to specify the location of the file, like you do in sqlite.
-        Keep in mind that in Windows you will have to use double backslashes '\\'
-        instead of forward slashes '/' when defining the file path.
+_SUBSCRIPT_TYPE = Union[Literal["__all__"], int, Set[str]]
+_SUBSCRIPT_DICT = Dict[str, Union[_SUBSCRIPT_TYPE, Literal["__none__"]]]
 
-    """
 
-    def __init__(self, path: str):
+class DBContextStorage(ABC):
```

Comment:
Maybe we should define concrete method wrappers for abstract ones.
E.g. `_load_main_info` is the abstract method, `load_main_info` -- concrete wrapper that calls `_load_main_info`.

This would simplify and standardize concrete methods (no need to use wrappers in concrete implementations, standardize logging, ensure db initialization, check field names):
```python
class DBContextStorage():
    @abc.abstractmethod
    async def _load_field_items(self, ctx_id, field_name):
        ...

    @DBContextStorage.lock
    async def load_field_items(self, ctx_id, field_name):
        logger.debug("Loading field items")
        self.validate_field_name(field_name)
        if not self.connected:
            await self.connect()
        
        return await self._load_field_items(ctx_id, field_name)

class ConcreteContextStorage(DBContextStorage):
    async def _load_field_items(self, ctx_id, field_name):
        # no field name validation here
        # no logging here
        return await self.db.load(ctx_id, field_name, self.subscript)
```